MQ2Medley.cpp

```diff
@@ -444,54 +556,144 @@ void Load_MQ2Medley_INI(PCHARINFO pCharInfo)
+		// Additional check to ensure there's a number after "swap"
+		std::string numberPart = type.substr(4); // Extract the part after "swap"
+		return !numberPart.empty() && std::all_of(numberPart.begin(), numberPart.end(), ::isdigit);
+	}
+	return false;
+}
+
 void Load_MQ2Medley_INI_Medley(PCHARINFO pCharInfo, const std::string& medleyNameIni)
 {
 	char szTemp[MAX_STRING] = { 0 };
 	char *pNext;
-
+	swapSettings.clear();
 	medley.clear();
 	Update_INIFileName(pCharInfo);
 
 	std::string iniSection = "MQ2Medley-" + medleyNameIni;
-	for (int i = 0; i < MAX_MEDLEY_SIZE; i++)
-	{
-		std::string iniKey = "song" + std::to_string(i + 1);
-		if (GetPrivateProfileString(iniSection.c_str(), iniKey.c_str(), "", szTemp, MAX_STRING, INIFileName))
-		{
+	WriteChatf("DEBUG::IniSection %s", iniSection.c_str());
+	std::map<std::string, std::string> conditions;
+	// Parse swap settings
+	int swapIndex = 1; // Start with swap1
+	while (true) {
+		std::string swapKey = "swap" + std::to_string(swapIndex);
+		//if (DebugMode) { WriteChatf("DEBUG::Attempting to read: %s", swapKey.c_str()); }
+		if (GetPrivateProfileString(iniSection.c_str(), swapKey.c_str(), "", szTemp, MAX_STRING, INIFileName)) {
+			if (DebugMode) { WriteChatf("DEBUG::Read %s: %s", swapKey.c_str(), szTemp); }
+			char *pItem = strtok_s(szTemp, "|", &pNext);
+			while (pItem) {
+				char *pSlot = strtok_s(nullptr, "|", &pNext);
+				if (pItem && pSlot) {
+					swapSettings[swapKey].emplace_back(pItem, pSlot);
+				}
+				pItem = strtok_s(nullptr, "|", &pNext); // Move to next pair
+			}
+			swapIndex++; // Move to the next swap
+		}
+		else {
+			if (DebugMode) { WriteChatf("DEBUG::No entry found for %s", swapKey.c_str()); }
+			break; // Exit the loop if no more swap settings are found
+		}
+	}
+	if (DebugMode) { WriteChatf("DEBUG::Total Swap Count: %d", swapIndex - 1); }
+
+
+	// Parse conditions dynamically
+	int condIndex = 1; // Start with cond1
+	while (true) {
+		std::string conditionKey = "cond" + std::to_string(condIndex);
+		if (GetPrivateProfileString(iniSection.c_str(), conditionKey.c_str(), "", szTemp, MAX_STRING, INIFileName)) {
+			if (DebugMode) { WriteChatf("DEBUG::Read %s: %s", conditionKey.c_str(), szTemp); }
+			conditions[conditionKey] = szTemp;
+			condIndex++; // Move to the next condition
+		}
+		else {
+			break; // Exit the loop if no more condition settings are found
+		}
+	}
+	if (DebugMode) { WriteChatf("DEBUG::Total Condition Count: %d", condIndex - 1); }
+
+	// Parse songs
+	for (int i = 0; i < MAX_MEDLEY_SIZE; i++) {
+		std::string songKey = "song" + std::to_string(i + 1);
+		if (GetPrivateProfileString(iniSection.c_str(), songKey.c_str(), "", szTemp, MAX_STRING, INIFileName)) {
 			SongData medleySong = nullSong;
 
-			//ugly ass split logic, example: song1=War March of Jocelyn^180.0^${Melee.Combat}
-			char *p = strtok_s(szTemp, "^", &pNext);
-			if (p)
-			{
+			// Determine the separator
+			char separator = strchr(szTemp, '|') ? '|' : '^';
```

Comment:
you are changing the config format here?  This is going to require everybody to update their config, right?

Reply:
I made it backward compatible so it will still function as normal for old configs. it checks for which separator is in use and will parse the info accordingly.